### PR TITLE
10964 move event presets

### PR DESCRIPTION
**Pull request by ablepharus**
## What does this PR do?

Close:  Move event presets to time line and remove events plugin (Dep: Separator widget) #10964 

## How was this PR tested?

Manual Testing in Animation Editor

image

### CODE REVIEW

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventPresetsWidget.cpp [hunk 1]
```diff
@@ -409,4 +424,86 @@ namespace EMStudio
...
+                createMotionEventCommand->SetEventDatas(preset->GetEventDatas());
+
+                if (!EMStudio::GetCommandManager()->ExecuteCommand(createMotionEventCommand, result))
+                {
+                    AZ_Error("EMotionFX", false, result.c_str());
+                }
+            }
+        }
+    }
+    bool MotionEventPresetsWidget::CheckIfIsPresetReadyToDrop()
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/MotionEvents/MotionEventPresetsWidget.cpp:485]: Nit: New line between functions

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp [hunk 2]
```diff
@@ -183,14 +184,33 @@ namespace EMStudio
...
+        // │  │            contentLayout                │    │                            │
+        // │  │ ┌─────────────────────────────────────┐ │    │                            │
+        // │  │ │         trackAndTrackDataLayout     │ │    │     motionEventsPresets    │
+        // │  │ ├─────────────────────────────────────┤ │    │                            │
+        // │  │ │                                     │ │Splitter                         │
+        // │  │ │          bodyWidget                 │ │    │                            │
+        // │  │ └─────────────────────────────────────┘ │    │                            │
+        // │  │                                         │    │                            │
+        // │  └─────────────────────────────────────────┘    │                            │
+        // └─────────────────────────────────────────────────┴────────────────────────────┘
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:210]: Nice!

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp [hunk 3]
```diff
@@ -207,27 +227,50 @@ namespace EMStudio
...
         bodyWidget->setWidget(innerWidget);
-        mainLayout->addWidget(bodyWidget, 2, 0, 1, 2);
 
         // Bottom-left
         m_trackHeaderWidget = new TrackHeaderWidget(this, m_dock);
         m_trackHeaderWidget->setFixedWidth(175);
         bodyLayout->addWidget(m_trackHeaderWidget);
 
         // Left
+        auto trackAndTrackDataWidget = new QWidget;
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:237]: Nit: `auto trackAndTrackDataWidget = new QWidget;` -> `auto* trackAndTrackDataWidget = new QWidget();`

Same for the other places below.

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp [hunk 4]
```diff
@@ -2036,6 +2072,24 @@ namespace EMStudio
...
     }
 
+    void TimeViewPlugin::toggleMotionEventPresetsPane()
+    {
+        QList<int> sizes = m_paneSplitter->sizes();
+        if (sizes[1] == 0)
+        {
+            sizes[1] = m_motionEventsPresetsWidget->minimumSizeHint().width();
+            sizes[0] = m_paneSplitter->width() - sizes[1];
+        }
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:2082]: Is the splitter grab- and resizeable as well?

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.h [hunk 5]
```diff
@@ -201,12 +206,16 @@ namespace EMStudio
...
+        TrackDataHeaderWidget*      m_trackDataHeaderWidget;
+        TrackDataWidget*            m_trackDataWidget;
+        TrackHeaderWidget*          m_trackHeaderWidget;
+        TimeInfoWidget*             m_timeInfoWidget;
+        TimeViewToolBar*            m_timeViewToolBar;
+        MotionEventPresetsWidget*   m_motionEventsPresetsWidget;
+        QWidget*                    m_mainWidget;
+        QSplitter*                  m_paneSplitter; // motion presets pane
+
+        QAction*                    m_togglePresetsView;
```
Comment (amzn-jillich) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.h:218]: Can you initialize these with nullptr please?

**Review by amzn-jillich — APPROVED** [review]

**Comment by amzn-jillich** [comment]
Just kicked off AR as well.

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp (hunk 4 above)
Comment (ablepharus) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:2082]: yes, it's freely resizeable. this is just for the toggle button above

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.h (hunk 5 above)
Comment (ablepharus) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.h:218]: sure thing

**Comment by amzn-jillich** [comment]
AR fails due to a couple of these (https://jenkins.build.o3de.org/job/O3DE/job/PR-11620/1/display/redirect):
`TimeViewPlugin.cpp::UnicodeValidator:198,85 FAILED - Source file contains unicode character, replace with \u2500.`
@ablepharus

**Comment by amzn-jillich** [comment]
DCO also failed, please see https://github.com/o3de/o3de/pull/11620/checks?check_run_id=8075274885 @ablepharus

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp (hunk 2 above)
Comment (hultonha) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:210]: Haha yeah that's some impressive ascii art skills 😅

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp [hunk 6]
```diff
@@ -2081,4 +2135,85 @@ namespace EMStudio
...
+                createMotionEventCommand->SetEventTrackName(eventTrack->GetName());
+                createMotionEventCommand->SetStartTime(aznumeric_cast<float>(dropTimeInSeconds));
+                createMotionEventCommand->SetEndTime(aznumeric_cast<float>(dropTimeInSeconds));
+                createMotionEventCommand->SetEventDatas(preset->GetEventDatas());
+
+                if (!EMStudio::GetCommandManager()->ExecuteCommand(createMotionEventCommand, result))
+                {
+                    AZ_Error("EMotionFX", false, result.c_str());
+                }
+            }
```
Comment (hultonha) [inline Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:2216]: suggestion: This looks very similar to code in `MotionEventPresetsWidget::OnEventPresetDropped`, any chance we can unify this (move to a function?)

**Review by hultonha — APPROVED** [review]
Overall looks great just a couple of suggestions. Thanks!

**Comment by hultonha** [comment]
Before/after image would be super helpful too for context (and a recording/video for these types of visual things are great too)

File: Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/CardHeader.cpp [hunk 7]
```diff
@@ -214,6 +214,7 @@ namespace AzQtComponents
     void CardHeader::setExpandable(bool expandable)
     {
         m_expanderButton->setEnabled(expandable);
+        m_expanderButton->setVisible(expandable);
```
Comment (cgalvan) [inline Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/CardHeader.cpp:217]: Just want to confirm with UX that we want to show/hide the expander button for cards if it's expandable or not? @amzn-leenguy @rainbj

Comment (amzn-leenguy) [reply Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/CardHeader.cpp:217]: Hey @cgalvan , Thanks or the tag. might need more context in the form like @hultonha suggested in recording of video or before after images. So i can give better feedback. 
Thanks!

Comment (hultonha) [reply Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/CardHeader.cpp:217]: @ablepharus could you please attach a recording showing of the changes in action please? Thanks!

Comment (ablepharus) [reply Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/CardHeader.cpp:217]: I grepped the code, `setExpandable` was never called with `false` and the `m_expanderButton` is enabled by default.
So, there seems to be no change to existing widgets.

If you, as i do in the other commit, set a CardHeader as not expandable, the expander icon (that will not work) will be hidden.

**Comment by ablepharus** [comment]
> AR fails due to a couple of these (https://jenkins.build.o3de.org/job/O3DE/job/PR-11620/1/display/redirect): `TimeViewPlugin.cpp::UnicodeValidator:198,85 FAILED - Source file contains unicode character, replace with \u2500.` @ablepharus

i guess this is the ascii diagramm. will remove it then

**Comment by ablepharus** [comment]
[10964-motion-event-presets.webm](https://user-images.githubusercontent.com/57981/188502085-34807c34-cbb3-412c-96bb-bdca5fde9d6e.webm)

File: Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/CardHeader.cpp (hunk 7 above)
Comment (ablepharus) [reply Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/CardHeader.cpp:217]: The Preset pane would look like this
image

File: Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp (hunk 6 above)
Comment (ablepharus) [reply Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp:2167]: Good catch!
The one in TimeViewPlugin can stay and do the work. I deleted the one in MotionEventPresets widget and connected the events to it.

**Comment by ablepharus** [comment]
I also added a commit that brings the same behaviour to anim graph pane. The toggle button is checked/unchecked depending on the visibility of the pane.

**Comment by hultonha** [comment]
Sounds good @ablepharus, let me kick off a build now 👍

**Comment by hultonha** [comment]
@ablepharus could you please update this branch with development and then I'll start another build

**Comment by amzn-jillich** [comment]
Just kicked off another build

**Comment by amzn-jillich** [comment]
Looks good to me! We need one more reviewer to approve it and get AR green still.

**Review by cgalvan — APPROVED** [review]

**Comment by cgalvan** [comment]
AR failed due to these warnings as errors:

```
[2022-09-19T14:53:04.574Z] D:/workspace/o3de/Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp(99,54): error C2220: the following warning is treated as an error [D:\workspace\o3de\build\windows\External\EMotionFX-78fb240e\Code\EMotionFX.Editor.Static.vcxproj]
[2022-09-19T14:53:04.574Z]       void TimeViewPlugin::OnBeforeRemovePlugin(uint32 classID)
[2022-09-19T14:53:04.574Z]                                                        ^
[2022-09-19T14:53:04.574Z] D:/workspace/o3de/Gems/EMotionFX/Code/EMotionFX/Tools/EMotionStudio/Plugins/StandardPlugins/Source/TimeView/TimeViewPlugin.cpp(99,54): warning C4100: 'classID': unreferenced formal parameter [D:\workspace\o3de\build\windows\External\EMotionFX-78fb240e\Code\EMotionFX.Editor.Static.vcxproj]
[2022-09-19T14:53:04.574Z]       void TimeViewPlugin::OnBeforeRemovePlugin(uint32 classID)
```

Once these are resolved I think this PR should be in good shape!

**Comment by hultonha** [comment]
@ablepharus Let me know when the reflection issue is solved and I can kick off another build, just give me a poke when it's ready 👍

**Pull request merged**